renderer/components/UI/LightningInvoiceInput.js

```diff
@@ -30,11 +30,31 @@ class LightningInvoiceInput extends React.Component {
       chainName += ` (${network})`
     }
 
+    // Ensure we have a value.
     if (isRequired && (!value || value.trim() === '')) {
       return intl.formatMessage({ ...messages.required_field })
     }
-    if (value && !isLn(value, chain, network) && !isOnchain(value, chain, network)) {
-      return intl.formatMessage({ ...messages.invalid_request }, { chain: chainName })
+
+    if (value) {
+      const invoiceIsLn = isLn(value, chain, network)
+      const invoiceIsOnchain = isOnchain(value, chain, network)
+
+      // Ensure we have a valid invoice or address.
+      if (!invoiceIsLn && !invoiceIsOnchain) {
+        return intl.formatMessage({ ...messages.invalid_request }, { chain: chainName })
+      }
+
+      // If we have a LN invoice, ensure the invoice has an amount.
+      if (invoiceIsLn) {
+        try {
+          const invoice = decodePayReq(value)
+          if (!invoice || (!invoice.satoshis || !invoice.millisatoshis)) {
+            throw new Error('Invalid invoice')
+          }
+        } catch (e) {
+          return intl.formatMessage({ ...messages.invalid_request }, { chain: chainName })
```

Comment:
if `decodePayReq` throws, won't this cause some UX confusion ? maybe it's better to just explicitly return 
```intl.formatMessage({ ...messages```
 from 
```if (!invoice || (!invoice.satoshis || !invoice.millisatoshis)) {``` 
block

Reply:
> if decodePayReq throws, won't this cause some UX confusion

The Error that gets thrown is caught below and handled by returning the appropriate error message, so the UI will behave as expected (show error if the invoice or invoice amount is invalid)